File: abc161/b/answer.py
N, M = map(int, input().split())
votes = list(map(int, input().split()))
min_votes = sum(votes) * 1.0 / (4*M)
votes_greater_than_min = list(filter(lambda x: x >= min_votes, votes))
if len(votes_greater_than_min) >= M:
    print('Yes')
else:
    print('No')

File: abc161/b/answer.kt
fun main(args: Array<String>) {
    val inputList1 = readLine()!!.split(' ').map { it.toInt() }
    val votes = readLine()!!.split(' ').map { it.toInt() }
    // val typeCountN = inputList1[0]
    val popularCountM = inputList1[1]
    val sumOfVotes = votes.sum()
    val validVotes = votes.filter { it >= sumOfVotes * 1.0 / (4*popularCountM) }
    // println(validVotes)
    // val validVotes = arrayOf<Int>()
    // for (vote in votes) {
    //     if (vote > sumOfVotes / (4*popularCountM))
    //         validVotes = validVotes + vote  // varでないので連結できていない plusAssign?
    // }
    if (validVotes.size >= popularCountM)
        println("Yes")
    else
        println("No")
}
